src/Impostor.Server/Net/Inner/Objects/InnerPlayerInfo.cs

```diff
@@ -215,5 +215,44 @@ private void SetTasks(ReadOnlyMemory<byte> taskTypeIds)
                 Tasks.Add(new TaskInfo(this, _eventManager, taskId++, taskType));
             }
         }
+
+        public byte GetNextRpcSequenceId(RpcCalls rpc)
+        {
+            var defaultOutfit = this.Outfits[PlayerOutfitType.Default];
+
+            byte nextRpcSequenceId = 0;
+
+            // ReSharper disable once SwitchStatementHandlesSomeKnownEnumValuesWithDefault
+            switch (rpc)
+            {
+                case RpcCalls.SetHatStr:
+                    ++defaultOutfit.HatSequenceId;
```

Comment:
Is a +1 sufficient or should we add, say, +5 to make sure our write wins from a concurrent write from the client?

Reply:
> Is a +1 sufficient or should we add, say, +5 to make sure our write wins from a concurrent write from the client?

I've been running this PR all day on skeld.net and not occured any issues so far.

Reply:
That doesn't answer my question. Players and the server may be racing to set a hat: if a player is setting a hat at the same time the server is setting a hat, who should win? If both of them send a hat at the same time, I think the server and the player will be desynced. Also applies to visors etc

Example from the Among Us source code: when RpcSnapTo is called, it adds 2 to the sequence id to make sure it wins from a player movement.